packages/frontend/editor-ui/src/features/dataStore/composables/useDataStorePagination.ts

```diff
@@ -0,0 +1,49 @@
+import { ref } from 'vue';
+
+export type UseDataStorePaginationOptions = {
+	initialPage?: number;
+	initialPageSize?: number;
+	pageSizeOptions?: number[];
+	onChange?: (page: number, pageSize: number) => Promise<void> | void;
+};
+
+export const useDataStorePagination = (options: UseDataStorePaginationOptions = {}) => {
+	const currentPage = ref<number>(options.initialPage ?? 1);
+	const pageSize = ref<number>(options.initialPageSize ?? 20);
+	const totalItems = ref<number>(0);
+	const pageSizeOptions = options.pageSizeOptions ?? [10, 20, 50];
+
+	const setTotalItems = (count: number) => {
+		totalItems.value = count;
+	};
+
+	const setCurrentPage = async (page: number) => {
+		currentPage.value = page;
+		if (options.onChange) await options.onChange(currentPage.value, pageSize.value);
+	};
+
+	const setPageSize = async (size: number) => {
```

Comment:
Since we are using fixed set of options for page sizes, maybe we can create type and use it instead of `number` here